src/mpg123/mpg123.cc

```diff
@@ -265,8 +267,14 @@ static bool read_mpg123_info(const char * filename, VFSFile & file,
         {
             tuple.set_int(Tuple::Length, length);
             tuple.set_int(Tuple::Bitrate, aud::rdiv<int64_t>(8 * size, length));
+            if (aud::rdiv<int64_t>(8 * size, length) !=  s.info.bitrate) vbr = MPG123_VBR;
         }
     }
+    
+    if (s.info.vbr != MPG123_CBR) vbr = s.info.vbr;
```

Comment:
As previously discussed, MPG123 only accurately detects VBR scenario when Xing tag is present ahead of the first frame.  Accordingly I don't think this logic is robust.

Reply:
Additionally, the variable assignment should be on a new line.

Reply:
That's why it is only used when not cbr. Otherwise the bitrate in info is compared to the calculated bitrate to determine vbr. (yes there is a minor chance that it will be incorrect but lt when xing header is not present). 

Variable assignment will I change as you wish.

Reply:
But my point is, MPG123 tends to report CBR mode for any file lacking a Xing tag ahead of the first frame.

Reply:
Yes, that's correct, that's why I only use the MPG123 info.vbr when it is not CBR because only then we know it's valid. Otherwise I use the comparison between calculated bitrate and bitrate reported by MPG123 (if they are equal it's very likely CBR otherwise it is VBR, and I never seen ABR before so i think we can leave that behind.

No it's not 100% waterproof, but I think that's not possible unless the whole file is scanned (all frames same size = cbr, different frame sizes = vbr) which takes more time.

Reply:
> Additionally, the variable assignment should be on a new line.

Done, I pushed a new commit.